cmd/abapEnvironmentPullGitRepo.go

```diff
@@ -154,3 +145,82 @@ func triggerPull(repositoryName string, pullConnectionDetails abaputils.Connecti
 	uriConnectionDetails.URL = body.Metadata.URI + expandLog
 	return uriConnectionDetails, nil
 }
+
+func checkPullRepositoryConfiguration(options abapEnvironmentPullGitRepoOptions) error {
+	if len(options.RepositoryNames) > 0 && len(options.Repositories) > 0 {
+		log.Entry().Info("It seems like you have specified both the repositories to be pulled as an in-line configuration as well as in the dedicated repositories configuration file.")
+		log.Entry().Info("Please note that in this case the dedicated repositories configuration file will be handled with priority.")
+	}
+	if len(options.RepositoryNames) == 0 && len(options.Repositories) == 0 {
+		return fmt.Errorf("Checking configuration failed: %w", errors.New("You have not specified any repository configuration to be pulled into the ABAP Environment System. Please make sure that you specified the repositories that should be pulled either in a dedicated file or via in-line configuration. For more information please read the User documentation"))
+	}
+	return nil
+}
+
+func pullReposFromConfigFile(repositoriesFile string, checkoutConnectionDetails abaputils.ConnectionDetailsHTTP, client piperhttp.Sender, pollIntervall time.Duration) (err error) {
```

Comment:
The responsibility of this function is not well defined. Why does it call finishPullLogs()? That should happen at the caller. Then you can reverse the if.

Reply:
I think it will not even be correct to finish the pull log here, since from the code-flow, I don't see how it cannot happen that pulling starts again in line 81.

Reply:
This function has also been removed. See above.